examples/terraform-aws-ssm-example/main.tf

```diff
@@ -0,0 +1,52 @@
+provider "aws" {
+  region = "us-east-2"
+}
+
+resource "aws_iam_role" "example" {
+  name = "example"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        }
+        Effect = "Allow"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "example-ssm" {
+  role       = aws_iam_role.example.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2RoleforSSM"
+}
+
+resource "aws_iam_instance_profile" "example" {
+  name = "example"
+  role = aws_iam_role.example.name
+}
+
+resource "aws_instance" "example" {
+  ami                         = "ami-04100f1cdba76b497"
```

Comment:
Please use the `aws_ami` data source to find the proper AMI ID for the region. Also, give the data source a name so we know what AMI this is! There are several examples in the repo of this you can copy.